internal/api/middleware/rbac.go

```diff
@@ -0,0 +1,96 @@
+package middleware
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"io"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/types"
+	"github.com/sirupsen/logrus"
+)
+
+var cfg *config.Config = config.GetConfig()
+var log *logrus.Logger = logging.GetLogger()
+
+func Rbac(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		permissions := get_user_permissions(c.Request().Header.Get("X-Rh-Identity"))
+		if permissions != nil {
+			c.Set("user.permissions", permissions)
+		} else {
+			return echo.NewHTTPError(http.StatusUnauthorized, "User is not authorized")
+		}
+		return next(c)
+	}
+}
+
+func aggregate_permissions(acls []types.RbacData) map[string][]string {
```

Comment:
Will it be possible to add doc string with short description what this method does?

Reply:
Added the doc string.